src/components/blocks/ExerciseSetBlock.tsx

```diff
@@ -101,13 +106,14 @@ const SetBlock: React.FC<SetBlockProps> = ({
           <TextInput
             accessibilityLabel="User Recorded Weight"
             accessibilityHint="Input weight you did"
-            className="my-auto mx-auto ml-1 h-5/6 w-16 rounded-md bg-white text-center font-bebas text-xs"
+            className="mx-auto my-auto ml-1 h-5/6 w-16 rounded-md bg-white text-center font-bebas text-xs"
             clearTextOnFocus={true}
             autoCapitalize="none"
             placeholder="Weight"
             returnKeyType="next"
             value={weight?.toString()}
             onChangeText={(value) => onChange(index, { weight: value })}
+            editable={workoutKey == currentWorkoutKey}
           />
         )}
         {bodyweight ? <View /> : <Text className="my-auto font-bebas text-xs font-bold"> LB</Text>}
```

Comment:
Also lock the checkbox when we are viewing an old workout. Right now, if you click the bodyweight checkbox while viewing past workout, it will change the current workout.